src/generators/schema-generator.php

```diff
@@ -180,6 +181,39 @@ private function get_type_from_piece( $piece ) {
 		return [];
 	}
 
+	/**
+	 * Validates a graph piece's type.
+	 *
+	 * When the type is an array:
+	 *   - Ensure the values are unique.
+	 *   - Only 1 value? Use that value without the array wrapping.
+	 *
+	 * @param array $piece The graph piece.
+	 *
+	 * @return array The graph piece.
+	 */
+	private function validate_type( $piece ) {
+		if ( ! isset( $piece['@type'] ) ) {
+			// No type to validate.
+			return $piece;
+		}
+
+		if ( \is_array( $piece['@type'] ) ) {
+			/*
+			 * Ensure the types are unique.
+			 * Use array_values to reset the indices (e.g. no 0, 2 because 1 was a duplicate).
+			 */
+			$piece['@type'] = \array_values( \array_unique( $piece['@type'] ) );
+
+			// Use the first value if there is only 1 type.
+			if ( \count( $piece['@type'] ) === 1 ) {
+				$piece['@type'] = \reset( $piece['@type'] );
+			}
+		}
+
+		return $piece;
```

Comment:
Although the code below adds some duplication (`return $piece` is duplicated), I think that it is better to return early. This makes the flow easier to follow.

Ah, the array does not contain this entry? Return.
Ah, the entry is not an array? Return.

I think that makes more sense than:
If it is an array, do this first and in both cases return eventually.

```suggestion
		// If it is not an array, we can return immediately.
		if ( ! \is_array( $piece['@type'] ) ) {
			return $piece;
		}
		
		/*
		 * Ensure the types are unique.
		 * Use array_values to reset the indices (e.g. no 0, 2 because 1 was a duplicate).
		 */
		$piece['@type'] = \array_values( \array_unique( $piece['@type'] ) );

		// Use the first value if there is only 1 type.
		if ( \count( $piece['@type'] ) === 1 ) {
			$piece['@type'] = \reset( $piece['@type'] );
		}
		
		return $piece;
```